### PR TITLE
Fix #13529 input group compact border color issues

### LINKED ISSUE #13529: 自定义表单控件使用Input.Group组件的compact属性后表单错误提示样式错误
- [x] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### Version
3.11.1

### Environment
Chrome 70.0.3538.110

### Reproduction link
[Edit on CodeSandbox](https://codesandbox.io/s/jnqmyqy40w)

### Steps to reproduce
直接点击submit后出现表单错误样式，右侧border样式bug

### What is expected?
...

### What is actually happening?
...

### PR DESCRIPTION
**Pull request by morenyang**
Fix #13529 
Try to fix border color issues in Input Group compact mode

Style align with [Input Addition](https://ant.design/components/input-cn/#components-input-demo-addon)

Please see the fix version

- Fix last child has different border color 
    image

- Fix different border color between first child and other children.

    image
    image

- Adjacent inputs share **ONE** border in the Input Group
    image
    image
    image   

    **But this may cause animations run not smooth**
    image

First of all, thank you for your contribution! :-)

Please makes sure that these checkboxes are checked before submitting your pull request, thank you!

* [x] Make sure that you propose pull request to right branch: bugfix for `master`, feature for branch `feature`.
* [x] Make sure that you follow antd's [code convention](https://github.com/ant-design/ant-design/wiki/Code-convention-for-antd).
* [x] Run `npm run lint` and fix those errors before submitting in order to keep consistent code style.
* [x] Rebase before creating a pull request to keep commit history clear.
* [x] Add some descriptions and refer relative issues for you pull request.

Extra checklist:

**if** *isBugFix* **:**

  * [x] Make sure that you add at least one unit test for the bug which you had fixed.

**elif** *isNewFeature* **:**

  * [ ] Update API docs for the component.
  * [ ] Update/Add demo to demonstrate new feature.
  * [ ] Update TypeScript definition for the component.
  * [ ] Add unit tests for the feature.

### CODE REVIEW

**Comment by zy410419243** [comment]
Need upgrade to [fix lint problem](4e4c737e9b7bf62cbde729432f88279d983c7064).

**Comment by morenyang** [comment]
Pass the style lint but could not commit 😂
It shows me prettier found some error, but prettier works well on my codes

**Comment by zy410419243** [comment]
Just update, it had been fixed in https://github.com/ant-design/ant-design/commit/4e4c737e9b7bf62cbde729432f88279d983c7064

**Comment by zy410419243** [comment]
这里 `rebase` 更新就好了...

``` bash
git checkout master
git pull --rebase upstream master
git checkout input-group-compact-issues-fix
git rebase master
git push origin input-group-compact-issues-fix -f
```

https://github.com/ant-design/ant-design/pull/4700#issuecomment-275305452
https://segmentfault.com/a/1190000000736629

File: components/drawer/demo/form-in-drawer.md [hunk 1]
```diff
@@ -40,7 +40,7 @@ class DrawerForm extends React.Component {
     return (
       <div>
         <Button type="primary" onClick={this.showDrawer}>
-          <Icon type="plus"/> New account
+          <Icon type="plus" /> New account
```
Comment (zy410419243) [inline components/drawer/demo/form-in-drawer.md]: 这个可以不用提交，因为在 `master` 里已经提交过了，你只要 `rebase` 就行了。

**Comment by morenyang** [comment]
new issue??

**Comment by zy410419243** [comment]
有时候是这样的，这种时候就得找个地方改点什么再提交一遍...

**Comment by zombieJ** [comment]
Try to avoid use `z-index`: 
kapture 2018-12-17 at 17 20 08

**Comment by morenyang** [comment]
> Try to avoid use `z-index`:
> kapture 2018-12-17 at 17 20 08

Hi, `z-index` is also use in `.input-group .ant-input`
https://github.com/ant-design/ant-design/blob/master/components/input/style/mixin.less#L139
https://github.com/ant-design/ant-design/blob/master/components/input/style/mixin.less#L143

This property is used to prevent right-border being covered.

**Comment by zombieJ** [comment]
> Hi, z-index is also use in .input-group .ant-input

Hmmm... It seems currently code also has this bug.

**Comment by morenyang** [comment]
> Hmmm... It seems currently code also has this bug.

Fix a bug and cause another bug.

**Pull request merged**